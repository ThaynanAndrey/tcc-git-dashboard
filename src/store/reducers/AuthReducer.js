import { USER_LOGIN_SUCCESS, USER_LOGIN_ERROR  } from '../actions/types';

const initState = {
    user: undefined,
    accessToken: undefined,
    error: undefined
};
  
const authReducer = (state = initState, action) => {
    switch(action.type) {
        case USER_LOGIN_SUCCESS:
            console.log('User logged');
            return {
                ...state,
                user: action.user
            };
        case USER_LOGIN_ERROR:
            console.log("Error to log user");
            return {
                ...state,
                user: undefined,
                error: action.error
            }
        default:
          return state
    }
};
  
export default authReducer;