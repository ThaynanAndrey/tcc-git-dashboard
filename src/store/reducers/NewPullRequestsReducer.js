import { LOAD_PULL_REQUESTS_NO_PROJECT_SUCCESS, ADDED_PULL_REQUEST_SUCCESS, ADDED_PULL_REQUEST_ERROR,
    LOAD_EXTERNAL_PULL_REQUESTS_NO_PROJECT_SUCCESS, LOAD_EXTERNAL_PULL_REQUESTS_NO_PROJECT_ERROR,
    RESET_PULL_REQUESTS_NO_PROJECT } from '../actions/types';

export const initState = {
    pullRequestsNoProject: [],
    externalPullRequestsNoProject: [],
};
  
const newPullRequestsReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PULL_REQUESTS_NO_PROJECT_SUCCESS:
            return {
                ...state,
                pullRequestsNoProject: action.pullRequestsNoProject
            };
        case ADDED_PULL_REQUEST_SUCCESS:
            return {
                ...state,
                pullRequestsNoProject: action.pullRequestsNoProject,
                externalPullRequestsNoProject: action.externalPullRequestsNoProject
            };
        case ADDED_PULL_REQUEST_ERROR:
            return state;
        case LOAD_EXTERNAL_PULL_REQUESTS_NO_PROJECT_SUCCESS:
            return {
                ...state,
                externalPullRequestsNoProject: action.externalPullRequestsNoProject
            }
        case LOAD_EXTERNAL_PULL_REQUESTS_NO_PROJECT_ERROR:
            return {
                ...state,
                externalPullRequestsNoProject: []
            }
        case RESET_PULL_REQUESTS_NO_PROJECT:
            return {
                ...state,
                pullRequestsNoProject: [],
                externalPullRequestsNoProject: []
            }
        default:
          return state
    }
};
  
export default newPullRequestsReducer;