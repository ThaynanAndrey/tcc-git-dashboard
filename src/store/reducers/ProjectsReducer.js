import { LOAD_PROJECTS_SUCCESS, LOAD_PROJECTS_ERROR, CREATE_PROJECT_SUCCESS, CREATE_PROJECT_ERROR } from '../actions/types';

const initState = {
    projects: [],
    currentProject: undefined,
    error: undefined
};
  
const projectsReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PROJECTS_SUCCESS:
            console.log('Projects loaded');
            return {
                ...state,
                projects: action.projects
            };
        case LOAD_PROJECTS_ERROR:
            console.log("Error to load projects");
            return {
                ...state,
                projects: [],
                error: action.error
            };
        case CREATE_PROJECT_SUCCESS:
            console.log("Create new project!");
            return {
                ...state,
                projects: action.projects
            };
        case CREATE_PROJECT_ERROR:
            console.log("Error to create project!");
            return state;
        default:
          return state;
    }
};
  
export default projectsReducer;