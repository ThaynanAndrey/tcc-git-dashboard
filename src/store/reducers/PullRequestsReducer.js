import { LOAD_PROJECT_PULL_REQUESTS_SUCCESS, LOAD_PULL_REQUEST_SUCCESS, RESET_PROJECT_PULL_REQUESTS } from '../actions/types';

const initState = {
    projectPullRequests: [],
    selectedPullRequest: { responsavel: {}, repositorio: {} },
};
  
const pullRequestsReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PROJECT_PULL_REQUESTS_SUCCESS:
            console.log('get all pull requests from project!');
            return {
                ...state,
                projectPullRequests: action.projectPullRequests
            };
        case LOAD_PULL_REQUEST_SUCCESS:
            return {
                ...state,
                selectedPullRequest: action.selectedPullRequest
            }
        case RESET_PROJECT_PULL_REQUESTS:
            return {
                ...state,
                projectPullRequests: []
            }
        default:
          return state
    }
};
  
export default pullRequestsReducer;