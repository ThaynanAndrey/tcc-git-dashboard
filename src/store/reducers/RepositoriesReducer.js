import { LOAD_PROJECT_REPOSITORIES_SUCCESS, LOAD_REPOSITORIES_NO_PROJECT_SUCCESS, ADDED_REPOSITORY_SUCCESS,
    ADDED_REPOSITORY_ERROR, LOAD_EXTERNAL_REPOSITORIES, LOAD_EXTERNAL_REPOSITORIES_ERROR, RESET_PROJECT_REPOSITORIES,
    RESET_REPOSITORIES_NO_PROJECT } from '../actions/types';

export const initState = {
    projectRepositories: [],
    repositoriesNoProject: [],
    externalRepositories: [],
    msgExternalRepositories: undefined,
    error: undefined
};
  
const repositoriesReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PROJECT_REPOSITORIES_SUCCESS:
            return {
                ...state,
                projectRepositories: action.projectRepositories
            };
        case LOAD_REPOSITORIES_NO_PROJECT_SUCCESS:
            return {
                ...state,
                repositoriesNoProject: action.repositoriesNoProject
            }
        case ADDED_REPOSITORY_SUCCESS:
            return {
                ...state,
                repositoriesNoProject: action.repositoriesNoProject,
                externalRepositories: action.externalRepositories
            }
        case ADDED_REPOSITORY_ERROR:
            return {
                ...state,
                error: action.error
            }
        case LOAD_EXTERNAL_REPOSITORIES:
        case LOAD_EXTERNAL_REPOSITORIES_ERROR:
            return {
                ...state,
                externalRepositories: action.externalRepositories,
                msgExternalRepositories: action.msgExternalRepositories
            }
        case RESET_PROJECT_REPOSITORIES:
            return {
                ...state,
                projectRepositories: []
            }
        case RESET_REPOSITORIES_NO_PROJECT:
            return {
                ...state,
                repositoriesNoProject: [],
                externalRepositories: []
            }
        default:
          return state
    }
};
  
export default repositoriesReducer;