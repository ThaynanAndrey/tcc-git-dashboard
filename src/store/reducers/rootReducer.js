import { combineReducers } from 'redux';
import authReducer from './AuthReducer';
import pullRequestsReducer from './PullRequestsReducer';
import newPullRequestsReducer from './NewPullRequestsReducer';
import repositoriesReducer from './RepositoriesReducer';

const rootReducer = combineReducers({
    auth: authReducer,
    pullRequests: pullRequestsReducer,
    newPullRequests: newPullRequestsReducer,
    repositories: repositoriesReducer
});

export default rootReducer;