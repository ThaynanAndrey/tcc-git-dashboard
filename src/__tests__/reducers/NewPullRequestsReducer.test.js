import newPullRequestsReducer from '../../store/reducers/NewPullRequestsReducer';
import { initState } from '../../store/reducers/NewPullRequestsReducer';

describe('NewPullRequestsReducer', () => {
    it('should return the initial state', () => {
        const state = newPullRequestsReducer(undefined, {});
        
        expect(state).toMatchSnapshot();
        expect(state).toEqual(initState);
    });

    it('should handle LOAD_PULL_REQUESTS_NO_PROJECT_SUCCESS', () => {
        const pullRequestsNoProject = [ { name: "PR Name", date: "10/10/1010" } ];
        
        const action = {
            type: 'LOAD_PULL_REQUESTS_NO_PROJECT_SUCCESS',
            pullRequestsNoProject
        };
        
        const newState = {
            ...initState,
            pullRequestsNoProject
        };

        const state = newPullRequestsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle ADDED_PULL_REQUEST_SUCCESS', () => {
        const pullRequestsNoProject = [ { name: "PR Crazy Name", date: "10/10/1010" } ];
        const externalPullRequestsNoProject = [ { name: "PR External Crazy Name", date: "10/12/1201" } ];
        
        const action = {
            type: 'ADDED_PULL_REQUEST_SUCCESS',
            pullRequestsNoProject,
            externalPullRequestsNoProject
        };
        
        const newState = {
            ...initState,
            pullRequestsNoProject,
            externalPullRequestsNoProject
        };

        const state = newPullRequestsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle ADDED_PULL_REQUEST_ERROR', () => {
        const action = {
            type: 'ADDED_PULL_REQUEST_ERROR'
        };

        const state = newPullRequestsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(initState);
    });

    it('should handle LOAD_EXTERNAL_PULL_REQUESTS_NO_PROJECT_SUCCESS', () => {
        const externalPullRequestsNoProject = [ { name: "PR External Crazy Name", date: "10/12/1201" } ];
        
        const action = {
            type: 'LOAD_EXTERNAL_PULL_REQUESTS_NO_PROJECT_SUCCESS',
            externalPullRequestsNoProject
        };
        
        const newState = {
            ...initState,
            externalPullRequestsNoProject
        };

        const state = newPullRequestsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle LOAD_EXTERNAL_PULL_REQUESTS_NO_PROJECT_ERROR', () => {        
        const action = {
            type: 'LOAD_EXTERNAL_PULL_REQUESTS_NO_PROJECT_ERROR'
        };
        
        const newState = {
            ...initState,
            externalPullRequestsNoProject: []
        };

        const state = newPullRequestsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle RESET_PULL_REQUESTS_NO_PROJECT', () => {        
        const action = {
            type: 'RESET_PULL_REQUESTS_NO_PROJECT'
        };
        
        const newState = {
            ...initState,
            pullRequestsNoProject: [],
            externalPullRequestsNoProject: []
        };

        const state = newPullRequestsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });
});