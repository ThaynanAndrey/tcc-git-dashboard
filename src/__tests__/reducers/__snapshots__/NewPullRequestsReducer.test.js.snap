// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPullRequestsReducer should handle ADDED_PULL_REQUEST_ERROR 1`] = `
Object {
  "externalPullRequestsNoProject": Array [],
  "pullRequestsNoProject": Array [],
}
`;

exports[`NewPullRequestsReducer should handle ADDED_PULL_REQUEST_SUCCESS 1`] = `
Object {
  "externalPullRequestsNoProject": Array [
    Object {
      "date": "10/12/1201",
      "name": "PR External Crazy Name",
    },
  ],
  "pullRequestsNoProject": Array [
    Object {
      "date": "10/10/1010",
      "name": "PR Crazy Name",
    },
  ],
}
`;

exports[`NewPullRequestsReducer should handle LOAD_EXTERNAL_PULL_REQUESTS_NO_PROJECT_ERROR 1`] = `
Object {
  "externalPullRequestsNoProject": Array [],
  "pullRequestsNoProject": Array [],
}
`;

exports[`NewPullRequestsReducer should handle LOAD_EXTERNAL_PULL_REQUESTS_NO_PROJECT_SUCCESS 1`] = `
Object {
  "externalPullRequestsNoProject": Array [
    Object {
      "date": "10/12/1201",
      "name": "PR External Crazy Name",
    },
  ],
  "pullRequestsNoProject": Array [],
}
`;

exports[`NewPullRequestsReducer should handle LOAD_PULL_REQUESTS_NO_PROJECT_SUCCESS 1`] = `
Object {
  "externalPullRequestsNoProject": Array [],
  "pullRequestsNoProject": Array [
    Object {
      "date": "10/10/1010",
      "name": "PR Name",
    },
  ],
}
`;

exports[`NewPullRequestsReducer should handle RESET_PULL_REQUESTS_NO_PROJECT 1`] = `
Object {
  "externalPullRequestsNoProject": Array [],
  "pullRequestsNoProject": Array [],
}
`;

exports[`NewPullRequestsReducer should return the initial state 1`] = `
Object {
  "externalPullRequestsNoProject": Array [],
  "pullRequestsNoProject": Array [],
}
`;
